lib/Backend/Vault/Kv.hs

```diff
@@ -132,10 +129,13 @@ embedCatchClientErrorMaybe
   => Member (Error CofferError) r
   => IO a
   -> Sem r (Maybe a)
-embedCatchClientErrorMaybe io = embed (catch @ClientError (io <&> Left) (pure . Right . exceptionHandler)) >>=
-  \case Left l -> (pure . Just) l
-        Right (Just r) -> throw r
-        Right Nothing -> pure Nothing
+embedCatchClientErrorMaybe io = embed (Ex.try @ClientError io) >>= \case
+  Left err ->
+    case exceptionHandler err of
+      -- 404 happened
+      Nothing -> pure Nothing
+      Just err -> throw err
+  Right r -> (pure . Just) r
```

Comment:
Hmm actually, now that `exceptionHandler` is only really used in one place, I think it makes sense to inline it here.

There's no need to transform a `ClientError` into a `Maybe CofferError` and then `Maybe CorrerError` into `Member (Error CofferError) r => Sem r a`. We can do it all in one single step.

```suggestion
  Left (FailureResponse _ response) 
    | statusCode (responseStatusCode response) == 404 -> pure Nothing
  Left err -> throw $ ServantError err
  Right r -> pure $ Just r
```